src/Agent.Worker/ExecutionContext.cs

```diff
@@ -410,6 +410,7 @@ public void InitializeJob(Pipelines.AgentJobRequestMessage message, Cancellation
 
             // JobSettings
             JobSettings = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            JobSettings[WellKnownJobSettings.JobId] = message.JobId.ToString();
```

Comment:
Is this jobId guaranteed to be the same every time across different runs? My understanding was that it was the same for any given orchestration, but not necessarily for different runs, so won't this break caching across runs when the job id is different?

Could be that I'm just misunderstanding the goal here.

Reply:
As I understand it (@zachariahcox @alex-peck to confirm), JobIds are derived based on some hash of the job information and should not change unless that information changes. Certainly, during my testing the JobId did not change between plan executions.

Reply:
JobId will always be the same assuming the underlying pipeline didn't change. It takes a "hash" of the stage name, phase name, job name, and attempt number (note: it will change on different attempts of the same job)

Reply:
> note: it will change on different attempts of the same job

I think that probably means we should prefer a different approach here that isn't dependent on attempt number, right? We weren't taking this into account previously and I think could lead to some weird behavior where a retry takes much longer than the initial attempt

Reply:
Thanks for starting this conversation Danny. After the discussions, we decided to use the repos hash instead of job info. This has a lot of benefits like if you change a jobs repositories to match another, it will use the same workspace as the other job.